language-support/hs/bindings/src/DA/Ledger/Services/PackageManagementService.hs

```diff
@@ -0,0 +1,28 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE GADTs #-}
+
+module DA.Ledger.Services.PackageManagementService (uploadDarFile) where
+
+import Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService
+import Data.ByteString(ByteString)
+import DA.Ledger.GrpcWrapUtils
+import DA.Ledger.LedgerService
+import Network.GRPC.HighLevel.Generated
+
+uploadDarFile :: ByteString -> LedgerService (Either String ()) -- Unlike other services, no LedgerId is needed. (why?!)
```

Comment:
Can we get a haddock comment here that explains when you will get back `Left`?

Reply:
ok